File: apps/android/app/src/main/kotlin/com/nogeass/passalarm/domain/model/AlarmSession.kt
package com.nogeass.passalarm.domain.model

sealed class AlarmSession {
    data object Idle : AlarmSession()

    data class Ringing(
        val tokenId: Long,
        val planId: Long,
        val totalRings: Int,
        val intervalMin: Int,
        val currentRingIndex: Int = 1,
    ) : AlarmSession() {
        val isComplete: Boolean get() = currentRingIndex > totalRings
        val progressText: String get() = "$currentRingIndex/$totalRings"
    }
}

File: apps/ios/PassAlarm/Domain/Entity/AlarmSession.swift
import Foundation

struct AlarmSession: Equatable, Sendable {
    let planId: UUID
    let totalRings: Int
    let intervalMin: Int
    var currentRingIndex: Int
    var isRinging: Bool
    var nextRingAt: Date?

    var isComplete: Bool {
        currentRingIndex >= totalRings
    }

    var progressText: String {
        "\(currentRingIndex)/\(totalRings)"
    }
}
